lib/Net/Stripe.pm

```diff
@@ -1545,10 +1570,70 @@ method _post(Str $path, $obj?) {
     return $self->_make_request($req);
 }
 
+sub BUILD {
+    my ( $self, $args ) = @_;
+
+    my $message = "The API version %s (%s) is not supported by this version of Net::Stripe. " .
+    "This version only supports %s or before. " .
+    "Please download a version-appropriate branch at https://github.com/lukec/stripe-perl/branches.";
+
+    if (defined($self->api_version)) {
+        if ($self->api_version gt $_MAX_API_VERSION) {
+            die sprintf($message,
+                "requested",
+                $self->api_version,
+                $_MAX_API_VERSION,
+            );
+        }
+    } else {
+        my $account_api_version = $self->_get_account_api_version();
+        if ($account_api_version gt $_MAX_API_VERSION) {
+            die sprintf($message,
+                "for your Stripe account",
+                $account_api_version,
+                $_MAX_API_VERSION,
+            );
+        }
+    }
+}
+
+# i really wish that the Stripe API had a ping-like method so that we
+# do not have to perform an extraneous request simply to check the
+# Stripe-Version header on the response.
+# for now, we will retrieve the balance, without specifying the API
+# version and attempt to check the response header.
+method _get_account_api_version {
+    my $path = 'balance';
+    my $req = GET $self->api_base . '/' . $path;
+
+    $req->header( Authorization =>
+        "Basic " . encode_base64($self->api_key . ':'));
+
+    if ($self->debug_network) {
+        print STDERR "Sending to Stripe:\n------\n" . $req->as_string() . "------\n";
+    }
+
+    my $resp = $self->ua->request($req);
+
+    if ($self->debug_network) {
+        print STDERR "Received from Stripe:\n------\n" . $resp->as_string() . "------\n";
+    }
+
+    my $stripe_version = $resp->header('Stripe-Version');
+    die "failed to retrieve the Stripe API version for this account"
+        unless $stripe_version && $stripe_version =~ /^\d{4}-\d{2}-\d{2}$/;
+
+    return $stripe_version;
+}
+
 method _make_request($req) {
     $req->header( Authorization =>
         "Basic " . encode_base64($self->api_key . ':'));
 
+    if ($self->api_version) {
```

Comment:
Thanks @sherrardb this makes perfect sense now.

Reply:
glad i could clarify. does that resolve all of your concerns?

Reply:
It certainly does!